File: C-Abi-Bridge-Aot/Tests/CAbiBridgeAot.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcCryption/CryptoPqcMlKemUtilsTest.cs

using System.Security.Cryptography;

namespace michele.natale.Tests;

partial class CryptoPqcMlKemTest
{
  public static MLKemAlgorithm[] ToMLKemAlgorithm()
  {
    var a = MLKemAlgorithm.MLKem512;
    var b = MLKemAlgorithm.MLKem768;
    var c = MLKemAlgorithm.MLKem1024;

    return [a, b, c];
  }

  public static MLKemAlgorithm ToMLKemAlgorithm(
    MLKemParam param) => param switch
    {
      MLKemParam.Ml_Kem_512 => MLKemAlgorithm.MLKem512,
      MLKemParam.Ml_Kem_768 => MLKemAlgorithm.MLKem768,
      MLKemParam.Ml_Kem_1024 => MLKemAlgorithm.MLKem1024,
      _ => throw new Exception(),
    };


  public static MLKemParam FromMLKemAlgorithm(
    MLKemAlgorithm parameter)
  {
    ArgumentNullException.ThrowIfNull(parameter);

    if (parameter == MLKemAlgorithm.MLKem512) return MLKemParam.Ml_Kem_512;
    else if (parameter == MLKemAlgorithm.MLKem768) return MLKemParam.Ml_Kem_768;
    else if (parameter == MLKemAlgorithm.MLKem1024) return MLKemParam.Ml_Kem_1024;

    throw new ArgumentOutOfRangeException(nameof(parameter), $"{nameof(parameter)} has failded!");
  }
}

File: C-Abi-Bridge-Aot/Tests/CAbiBridge.Aot.Vb.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcCryption/CryptoPqcMlKemUtilsTest.vb
Option Strict On
Option Explicit On
Imports System.Security.Cryptography
Imports michele.natale

Namespace michele.natale.Tests
  Partial Class CryptoPqcMlKemUtilsTest
    Public Shared Function ToMLKemAlgorithm() As MLKemAlgorithm()
      Dim a = MLKemAlgorithm.MLKem512
      Dim b = MLKemAlgorithm.MLKem768
      Dim c = MLKemAlgorithm.MLKem1024

      Return {a, b, c}
    End Function

    Public Shared Function ToMLKemAlgorithm(param As MLKemParam) As MLKemAlgorithm
      Select Case param
        Case MLKemParam.Ml_Kem_512 : Return MLKemAlgorithm.MLKem512
        Case MLKemParam.Ml_Kem_768 : Return MLKemAlgorithm.MLKem768
        Case MLKemParam.Ml_Kem_1024 : Return MLKemAlgorithm.MLKem1024
        Case Else : Throw New Exception()
      End Select
    End Function

    Public Shared Function FromMLKemAlgorithm(parameter As MLKemAlgorithm) As MLKemParam
      ArgumentNullException.ThrowIfNull(parameter)

      If parameter Is MLKemAlgorithm.MLKem512 Then
        Return MLKemParam.Ml_Kem_512
      ElseIf parameter Is MLKemAlgorithm.MLKem768 Then
        Return MLKemParam.Ml_Kem_768
      ElseIf parameter Is MLKemAlgorithm.MLKem1024 Then
        Return MLKemParam.Ml_Kem_1024
      End If

      Throw New ArgumentOutOfRangeException(NameOf(parameter), $"{NameOf(parameter)} has failded!")
    End Function
  End Class
End Namespace
